### ISSUE TITLE
Support TRUNCATE table

### ISSUE THREAD

**Issue by kbajda** [prestodb/presto#3424]
Having something like TRUNCATE to quickly empty an existing table seems desirable (e.g., http://www.postgresql.org/docs/9.4/static/sql-truncate.html).

Note that "TRUNCATE table_name" works similarly to "DELETE FROM table_name" but particular connector may have a more efficient method to wipe out all data from a given table than running a generic DELETE operation that may accept predicates via WHERE clause.

**Comment by martint** [comment]
It's probably ok to have an efficient way in the connector API to do this, but I would prefer not adding specialized syntax. According to the ANS SQL spec: "Delete all rows of a base table without causing any triggered action", so given that we don't support triggers or side-effects yet we probably don't need it for now.

**Comment by electrum** [comment]
There an interesting question regarding the semantics for connectors. For example, when running against the PostgreSQL connector, a `DELETE FROM` and `TRUNCATE TABLE` might have different semantics on the PostgreSQL side.

**Comment by kokosing** [comment]
Here you have described differences between `DELETE FROM` and `TRUNCATE` in PostgreSQL - http://stackoverflow.com/questions/11419536/postgresql-truncation-speed/11423886#11423886 
@KBP-TDC  It looks like `DELETE FROM` with no predicate is more efficient there as it does not perform any `VACUUM` operations.

**Comment by findepi** [comment]
The `delete from` vs `truncate` in PostgreSQL is also explained in their docs https://www.postgresql.org/docs/9.6/static/sql-truncate.html

> TRUNCATE quickly removes all rows from a set of tables. It has the same effect as an unqualified DELETE on each table, but since it does not actually scan the tables it is faster. [...] This is most useful on large tables.

There are further constrains when foreign keys are in play (you can't truncate table _possibly_ referenced, but you can _delete from_ it unless there _really_ exists a referencing row).

In Oracle, for example, `truncate` is not transactional (so does not require copying table's data to "undo tablespace"). 

Bottom line is that different databases have support for `truncate` and it's often different operation than `delete from`.
